proxy/src/main/java/net/md_5/bungee/ServerConnector.java

```diff
@@ -325,7 +333,20 @@ public void handle(Login login) throws Exception
     @Override
     public void handle(EncryptionRequest encryptionRequest) throws Exception
     {
-        throw new QuietException( "Server is online mode!" );
+        byte[] secret = new byte[16];
+        ThreadLocalRandom.current().nextBytes( secret );
```

Comment:
This is no sufficient randomness for encryption.
Use SecureRandom.

Reply:
EncryptionResponse -> Configured Server directly